packit/api.py

```diff
@@ -623,24 +665,18 @@ def create_srpm(
         """
         Create srpm from the upstream repo
 
-        :param upstream_ref: git ref to upstream commit
-        :param output_file: path + filename where the srpm should be written, defaults to cwd
-        :param srpm_dir: path to the directory where the srpm is meant to be placed
-        :return: a path to the srpm
+        Args:
+            upstream_ref: git ref to upstream commit
+            output_file: path + filename where the srpm should be written, defaults to cwd
+            srpm_dir: path to the directory where the srpm is meant to be placed
+            release_suffix: specifies local release suffix. `None` represents default suffix.
+            bump_version: specifies whether version should be changed in the spec-file.
+
+        Returns:
+            a path to the srpm
         """
         try:
-            self.up.run_action(actions=ActionName.post_upstream_clone)
-
-            try:
-                self.up.prepare_upstream_for_srpm_creation(
-                    upstream_ref=upstream_ref,
-                    bump_version=bump_version,
-                    release_suffix=release_suffix,
-                )
-            except Exception as ex:
-                raise PackitSRPMException(
-                    f"Preparation of the repository for creation of an SRPM failed: {ex}"
-                ) from ex
+            self.prepare_sources(upstream_ref, bump_version, release_suffix)
```

Comment:
this is really awesome that you abstracted it to a separate method, much cleaner now